bee-api/bee-rest-api/src/types/dtos.rs

```diff
@@ -0,0 +1,954 @@
+/// which unlock through the same data.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ReferenceUnlockDto {
+    #[serde(rename = "type")]
+    pub kind: u8,
+    #[serde(rename = "reference")]
+    pub index: u16,
+}
+
+/// The payload type to define a milestone.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct MilestonePayloadDto {
+    #[serde(rename = "type")]
+    pub kind: u32,
+    pub index: u32,
+    pub timestamp: u64,
+    #[serde(rename = "parentMessageIds")]
+    pub parents: Vec<String>,
+    #[serde(rename = "inclusionMerkleProof")]
+    pub inclusion_merkle_proof: String,
+    #[serde(rename = "nextPoWScore")]
+    pub next_pow_score: u32,
+    #[serde(rename = "nextPoWScoreMilestoneIndex")]
+    pub next_pow_score_milestone_index: u32,
+    #[serde(rename = "publicKeys")]
+    pub public_keys: Vec<String>,
+    pub receipt: Option<PayloadDto>,
+    pub signatures: Vec<String>,
+}
+
+impl From<&MilestonePayload> for MilestonePayloadDto {
+    fn from(value: &MilestonePayload) -> Self {
+        MilestonePayloadDto {
+            kind: MilestonePayload::KIND,
+            index: *value.essence().index(),
+            timestamp: value.essence().timestamp(),
+            parents: value.essence().parents().iter().map(|p| p.to_string()).collect(),
+            inclusion_merkle_proof: hex::encode(value.essence().merkle_proof()),
+            next_pow_score: value.essence().next_pow_score(),
+            next_pow_score_milestone_index: value.essence().next_pow_score_milestone_index(),
+            public_keys: value.essence().public_keys().iter().map(hex::encode).collect(),
+            receipt: value.essence().receipt().map(Into::into),
+            signatures: value.signatures().iter().map(hex::encode).collect(),
+        }
+    }
+}
+
+impl TryFrom<&MilestonePayloadDto> for MilestonePayload {
+    type Error = Error;
+
+    fn try_from(value: &MilestonePayloadDto) -> Result<Self, Self::Error> {
+        let essence = {
+            let index = value.index;
+            let timestamp = value.timestamp;
+            let mut parent_ids = Vec::new();
+            for msg_id in &value.parents {
+                parent_ids.push(
+                    msg_id
+                        .parse::<MessageId>()
+                        .map_err(|_| Error::InvalidSyntaxField("parentMessageIds"))?,
+                );
+            }
+            let merkle_proof = {
+                let mut buf = [0u8; MILESTONE_MERKLE_PROOF_LENGTH];
+                hex::decode_to_slice(&value.inclusion_merkle_proof, &mut buf)
+                    .map_err(|_| Error::InvalidSyntaxField("inclusionMerkleProof"))?;
+                buf
+            };
+            let next_pow_score = value.next_pow_score;
+            let next_pow_score_milestone_index = value.next_pow_score_milestone_index;
+            let mut public_keys = Vec::new();
+            for v in &value.public_keys {
+                let key = {
```

Comment:
Why is this in its own scope?

Reply:
the idea was to make it more easy to understand that these opeartions belong to the key; is this more confusing?